### ISSUE TITLE
Surprising values returned by gp_minimize from categories

### ISSUE THREAD

**Issue by gar1t** [scikit-optimize/scikit-optimize#821]
7720fdac is an ostensible fix for #752 but the results from `gp_minimize` now return invalid category values.

Sample to recreate:

``` python
import skopt

def f(params):
    return 0

dims = [[1, 2, 3]]

res = skopt.gp_minimize(
    f,
    dims,
    n_calls=1,
    n_random_starts=1,
)

print(res.x_iters)
```

Output:

```
[['1']]
```

Note that the suggested value is a string value, which is not one of the category values.

The transform to strings is predicated on the ability to reverse transform the values, but this isn't happening here.

**Comment by gar1t** [comment]
Corrected the title as I might be breaking the abstraction here. If `x_iters` is an internal structure (from the docs, it's the _location of function evaluation for each iteration_) then it would be helpful to have the type information for use in a call to `space.inverse_transform()` to get the suggested values. As it is there's no way to know from the return value what values were used in calls to `func`.

**Comment by gar1t** [comment]
Hehe, I take it back, from what I'm seeing, this is extremely invalid :)

Here's clearer example:

```python
import skopt

def f(params):
    print(params)
    return 0

dims = [[1]]

res = skopt.gp_minimize(
    f,
    dims,
    n_calls=1,
    n_random_starts=1,
)
```

Output is always `['1']`.

**Comment by holgern** [comment]
The problem is that there is not an inverse_transform performed when the function f is called.
```
from skopt.space import Space, Categorical, Integer, Real, Dimension
space= Space([[1,2,3]])
dimension = space.dimensions[0]
dims = Categorical(dimension.categories,
                                dimension.prior,
                                name=dimension.name,
                                transform="identity")
print(dims)
# Categorical(categories=('1', '2', '3'), prior=None)
print(dims._rvs.rvs(size=1))
# 1
print(dims.rvs(1))
# ['2']
```
There seems to me a problem with a missing inverse_transform

**Comment by holgern** [comment]
Will be fixed in 0.7.1
